Use expected URL instead of expected origins for IAE flow (#695)
openid/OpenID4VCI: 1.1/openid-4-verifiable-credential-issuance-1_1.md

If `type` is set to `openid4vp_presentation`, as shown in the following example, the response MUST further include an `openid4vp_request` parameter containing an OpenID4VP Authorization Request. The contents of the request is the same as for requests passed to the Digital Credentials API (see Appendix A.2 and Appendix A.3 of [@!OpenID4VP]), except as follows:

* The `response_mode` MUST be either `iae_post` for unencrypted responses or `iae_post.jwt` for encrypted responses. These modes are used to indicate to the Wallet to return the response back to the same Interactive Authorization Endpoint.
* The `expected_origins` parameter MUST NOT be present.

> * When signed requests defined in A.3.2 of [@!OpenID4VP] are used, an addtional `expected_url` parameter MUST be present. The `expected_url` parameter is a non-empty string that contains the endpoint of the Verifier that is returning the request. In the `openid4vp_presentation` interaction type, `expected_url` MUST be set to the Interactive Authorization Endpoint.

Author: @c2bo is this better?
```suggestion
* When signed requests defined in A.3.2 of [@!OpenID4VP] are used, an additional `expected_url` parameter MUST be present. The `expected_url` parameter is a non-empty string that contains the endpoint of the Authorization Server that is returning the Interactive Authorization Response. In the `openid4vp_presentation` interaction type, `expected_url` MUST be set to the Interactive Authorization Endpoint.
```

When processing the request the following logic applies:

>   1. If `expected_url` is present, the Wallet MUST compare the value in this parameter to the URL of the follow-up request to detect replay of the request from a malicious Verifier. If the URL does not match value of the `expected_url`, the Wallet MUST return an error. This error SHOULD be an `invalid_request` error. This parameter is not for use in unsigned requests and therefore a Wallet MUST ignore this parameter if it is present in an unsigned request.

Reviewer: From reading this text, I'm not really sure what a wallet should do.

Before this change you were required to derive the origin from the IAE endpoint and compared this to the `expected_origins`, correct?

After this change you take the exact IAE endpoint and compare that against the `expected_url`? The text above mentions "string that contains the endpoint of the Verifier that is returning the request" which is a bit vague to me, and different from what was done before. Shouldn't it be the Issuer returning the request? And then the OID4VP request has a link to the issuer url (based on `expected_url`).

- Is it correct Verifier should be Issuer?
- Is it possible to define "endpoint of the Verifier that is returning the request" more clearly (is it the IAE endpoint?)

Author: I believe we agreed that we will create a dedicated issue for generic design. I will keep the parameter name generic though

Author: However, I made the language future-proof in case we forget to update this section once we made the section more generic.

>   1. If `expected_url` is present, the Wallet MUST compare the value in this parameter to the URL of the follow-up request to detect replay of the request from a malicious Verifier. If the URL does not match the value of the `expected_url`, the Wallet MUST return an error. This error SHOULD be an `invalid_request` error. This parameter is not for use in unsigned requests and therefore a Wallet MUST ignore this parameter if it is present in an unsigned request.

Author: @c2bo is this better?
```suggestion
  1. If `expected_url` is present, the Wallet MUST compare the value in this parameter to the URL of the follow-up Interactive Authorization Request to detect replay of the request from a malicious Authorization Server/Verifier. If the URL does not match the value of the `expected_url`, the Wallet MUST return an error. This error SHOULD be an `invalid_request` error. This parameter is not for use in unsigned requests and therefore a Wallet MUST ignore this parameter if it is present in an unsigned request.
```

Custom extensions ((#iae-custom-extensions)) MUST ensure that this attack is prevented by ensuring one or both of the following:

>  1. The Wallet is able to detect that a request is not presented by the party that initiated the Interactive Authorization Request. In the case of the (#iae-require-presentation) interaction with a signed Presentation request, this is achieved by the Wallet verifying the `expected_url` parameter in the request, which contains the Interactive Authorization Endpoint that initiated the request.

Author: @c2bo is this better?
```suggestion
 1. The Wallet is able to detect that a request is not presented by the party that initiated the Interactive Authorization Request. In the case of the (#iae-require-presentation) interaction with a signed Presentation request, this is achieved by the Wallet verifying the `expected_url` parameter in the Interactive Authorization Response, which contains the Interactive Authorization Endpoint that initiated the Interactive Authorization Response.
```
